primitives/src/merkle_tree/field_based_mht/smt/error.rs

```diff
@@ -0,0 +1,30 @@
+use std::{error, fmt};
+
+#[derive(Debug, Clone)]
+/// Possible error states for the hashing.
+pub enum Error {
```

Comment:
There is already a MerkleTreeError struct defined in primitives/src/merkle_tree/mod.rs. Merge this one with the pre-exisisting one